### PR TITLE
Created IClient interface

### PR DESCRIPTION
**Pull request by danma23**
I need to unit test my analytics intergration without hitting the Segment servers.
These changes add an IClient interface that is implemented by the Client class.

### CODE REVIEW

File: Analytics.Xamarin.Pcl/Client.cs [hunk 1]
```diff
@@ -30,13 +32,41 @@ public class Client : IDisposable
...
+			{
+				return _config;
+			}
+		}
+
+		#endregion
+
 		#region Initialization
 
+		public Client()
```
Comment (f2prateek) [inline Analytics.Xamarin.Pcl/Client.cs]: There shouldn't be an empty constructor. A writeKey must be provided.

File: Analytics.Xamarin.Pcl/IClient.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		#endregion //Properties
+
+		#region Methods
+
+		/// <summary>
+		/// Creates a new REST client with a specified API writeKey and default config
+		/// </summary>
+		/// <param name="writeKey"></param>
+		/// <param name="config"></param>
+		void Initialize(string writeKey, Config config = null);
```
Comment (f2prateek) [inline Analytics.Xamarin.Pcl/IClient.cs]: This shouldn't be part of the interface. Implementations can have their own constructor to accept this. E.g. in the testing your application, you don't really need the writeKey and config.

**Comment by f2prateek** [comment]
Looks good overall - I think we should try to make the Client class immutable as much as possible by keeping initialization in the constructor, and only expose the tracking methods in the interface.

**Comment by danma23** [comment]
Sure... I moved the Initialize method out of the IClient interface and made it a protected method. That way, the Client class is immutable but can still be unit tested without hitting the Segment backend :)

**Pull request closed without merging**